basiclti/basiclti-blis/src/java/org/sakaiproject/lti13/LTI13Servlet.java

```diff
@@ -295,6 +301,62 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
 
 	}
 
+	// A very locked down proxy - JSON Only
+	protected void handleProxy(HttpServletRequest request, HttpServletResponse response) {
+		String proxyUrl = request.getParameter("proxyUrl");
+		if ( proxyUrl == null ) {
+			LTI13Util.return400(response, "Missing proxyUrl");
+			return;
+		}
+
+		Session sess = SessionManager.getCurrentSession();
+		if ( sess == null || sess.getUserId() == null ) {
+			LTI13Util.return400(response, "Must be logged in");
+			return;
+		}
+
+		// https://stackoverflow.com/a/43708457/1994792
+		try {
+			java.net.URL url = new java.net.URL(proxyUrl);
+			java.net.HttpURLConnection con = (java.net.HttpURLConnection) url.openConnection();
+			con.setRequestMethod("GET");
+			con.setConnectTimeout(3000);
+			con.setReadTimeout(3000);
+			con.setInstanceFollowRedirects(true);
+
+			try ( java.io.BufferedReader in = new java.io.BufferedReader(
+				new java.io.InputStreamReader(con.getInputStream())) )
+			{
+				String inputLine;
+				StringBuffer content = new StringBuffer();
+				while ((inputLine = in.readLine()) != null) {
```

Comment:
I believe there's potential for a DoS attack here if the input just keeps coming. It would never make it to the check after. You'd either need to use a size limited BufferedReader or a `org.apache.commons.io.input.BoundedInputStream`. 

This suggests
https://stackoverflow.com/a/17142341/3708872

```
BufferedReader bufferedReader = new BufferedReader(
  new InputStreamReader(
    new BoundedInputStream(inputStream, <no-of-bytes>)
  )
);
```

Reply:
Oops I didn't see that this was merged **as** I was typing the comment.